resource/utils_test.go

```diff
@@ -0,0 +1,60 @@
+package resource
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestRedact(t *testing.T) {
+	secrets := map[string]struct{}{
+		"redactme!": {},
+	}
+
+	testCases := []struct {
+		name  string
+		dirty map[string]any
+		want  map[string]any
+	}{
+		{
+			name: "redact one entry",
+			dirty: map[string]any{
+				"redactme!": "the secret",
```

Comment:
Why the exclamation mark in `redactme!` ?

Reply:
To make it more evident; is this confusing somehow ?
What is important is that it matches a key in lines 10--12

Reply:
For a second or two, I asked myself if this with exclamation mark is some kind of Golang trick. Then I had a look and I could not find any other possibility except this being a simple string. So to answer is it confusing, not much, but I would prefer:

`secret_string_to_redact` or something equivalent i.e `SecretStringToRedact`

Reply:
The exclamation mark cannot be a Go trick (well, should not be a Go trick from the point of view of a decent API) because that string is part of the public API: any key set in `secrets` will have its value redacted. 

The `redact` documentation says:

```go
// redact makes a copy of dirty and, for each k/v pair of it, if k matches one of the
// keys in secrets AND v is a string, it redacts v. It returns the redacted copy.
// WARNING: it is able to redact only string values!
func redact(dirty map[string]any, secrets map[string]struct{}) map[string]any
```

But I can remove the `!`, it is not needed.

The thing with `secret_string_to_redact` is that it risks being even more confusing: it is not the string itself, it is the associated value. It should be something like `key_with_value_to_redact`, this is why I used `redactme`...

I didn't like using `secrets map[string]struct{}`, but Go doesn't have sets, so this is how to emulate a set.
I could also have used the more evident `secrets []string`, but then I would have had an uglier implementation.

If redact was part of the public API, then I should have used `secrets []string`. Since it is private, I went with implementer simplicity. Maybe all this discussion is telling me to use `secrets []string` in any case.

Reply:
Done